### PR TITLE
Prettier

### PR DESCRIPTION
**Pull request by buschtoens**
This PR adds the [Prettier](https://prettier.io/) code auto formatter as a fixable eslint rule.

This means that a uniform coding style will be enforced by linter rules. If you violate the coding style, ember-cli will scream warnings. You can also configure your editor to fix eslint errors on save, which I highly recommend. Doing so will ensure that your code is automatically formatted.

I disabled Prettier for some bits, as that could cause pain when upgrading ember-cli.

We adopted Prettier at work and I think it greatly improved our developer experience. 99 % the time it generates easier to read code. For the other 1% there's [`// prettier-ignore`](https://prettier.io/docs/en/usage.html#excluding-code-from-formatting) or `/* eslint-disable prettier/prettier */`.

I have no hard feelings, if you're against merging.

### CODE REVIEW

**Comment by alexander-alvarez** [comment]
It's definitely not a bad thing, although I wouldn't have done some of the formatting they codified.
I'm was waiting to see what comes of https://github.com/emberjs/core-notes/blob/master/ember-cli/2017-06/june-29.md#prettier but as always you're one step ahead.

**Comment by buschtoens** [comment]
Haha, yeah. I suffer from a real bad case of shiny new toy syndrome. :joy: 

It's true that there are a few places, where I would have styled differently, but the overall consistency outweighs these few for me.

Thank you for the link to the notes. Must have slipped my attention.

File: addon/components/lt-body.js [hunk 1]
```diff
@@ -336,7 +335,12 @@ export default Component.extend({
       _scrollTo,
       scrollToRow,
       _scrollToRow
-    } = this.getProperties(['scrollTo', '_scrollTo', 'scrollToRow', '_scrollToRow']);
+    } = this.getProperties([
+      'scrollTo',
+      '_scrollTo',
+      'scrollToRow',
+      '_scrollToRow'
+    ]);
```
Comment (buschtoens) [inline addon/components/lt-body.js]: For example, I wouldn't write these destructuring `getProperties` statements anymore. Instead I'd go with multiple `get` lines.

I'll change that, when I'm back in the office.

File: addon/components/lt-column-resizer.js [hunk 2]
```diff
@@ -21,7 +17,9 @@ export default Component.extend({
 
   $column: computed(function() {
     return $(this.get('element')).parent('th');
-  }).volatile().readOnly(),
+  })
+    .volatile()
+    .readOnly(),
```
Comment (buschtoens) [inline addon/components/lt-column-resizer.js]: That's also a bit weird. But that would go away by using `ember-decorators`.

File: addon/utils/css-styleify.js [hunk 3]
```diff
@@ -1,17 +1,11 @@
 import Ember from 'ember';
 
-const {
-  isPresent,
-  String: {
-    dasherize,
-    htmlSafe
-  }
-} = Ember;
+const { isPresent, String: { dasherize, htmlSafe } } = Ember;
```
Comment (buschtoens) [inline addon/utils/css-styleify.js]: This also looks a bit weird. But ES6 module imports fix this as well.

Which reminds me - we should upgrade the code base to ES6 module imports. :grin:

**Comment by fran-worley** [comment]
Given how out of date this now is, I'm going to close so we can focus on ember octane support.

**Pull request closed without merging**